cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/LoginAction.java

```diff
@@ -121,9 +126,17 @@ public void doAction() throws Exception {
             throw new AccessControlException("login failed");
         }
         
-        // TODO Alinga
-        // Add group check on rp.accessGroup here
-        // (will require client_id to be passed from AuthorizeAction, to oidc-login.html, to here)
+        // check client id
+        RelyParty rp = OIDCUtil.getRelyParty(clientID);
+        if (rp == null) {
+            throw new AccessControlException("login failed, unauthorized client " + clientID);
+        }
+        // perform group check on rp.accessGroup 
+        if (!OIDCUtil.accessAllowed(rp)) {
+            GroupURI accessGroup = rp.getAccessGroup();
+            String msg = "login failed, client " + clientID + " is not a member of " + accessGroup;
```

Comment:
same comment as in Authorize action.  Error should be: ""login failed, not a member of group <groupURI>"

Reply:
Right. Thanks.